projects/packages/publicize/src/class-social-admin-page.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Social Admin Page class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize;
+
+use Automattic\Jetpack\Admin_UI\Admin_Menu;
+use Automattic\Jetpack\Assets;
+
+/**
+ * The class to handle the Social Admin Page.
+ */
+class Social_Admin_Page {
+
+	/**
+	 * The instance of the class.
+	 *
+	 * @var Social_Admin_Page
+	 */
+	private static $instance;
+
+	/**
+	 * Initialize the class.
+	 *
+	 * @return Social_Admin_Page
+	 */
+	public static function init() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * The constructor.
+	 */
+	private function __construct() {
+		add_action( 'admin_menu', array( $this, 'add_menu' ) );
+
+		/**
+		 * Admin_Menu::add_menu uses 1000, so we use 2000
+		 * to ensure we remove the old Social menu item after it has been added.
+		 */
+		add_action( 'admin_menu', array( $this, 'remove_old_social_menu_item' ), 2000 );
+	}
+
+	/**
+	 * Remove the page added by old versions of Social.
+	 */
+	public function remove_old_social_menu_item() {
+
+		if ( defined( 'JETPACK_SOCIAL_PLUGIN_ROOT_FILE' ) ) {
+
+			$plugin_data = get_plugin_data( (string) constant( 'JETPACK_SOCIAL_PLUGIN_ROOT_FILE' ), false, false );
+
+			$plugin_version = $plugin_data['Version'];
```

Comment:
Can we get this from the script data function, or move that function somewhere else?

Reply:
Done. Thanks